model/converter/json/from_domain.go

```diff
@@ -55,29 +66,43 @@ func (fd fromDomain) fromDomain(trace *model.Trace) *json.Trace {
 	return jTrace
 }
 
-func (fd fromDomain) convertSpan(span *model.Span, processID json.ProcessID) json.Span {
+func (fd fromDomain) convertSpanInternal(span *model.Span) json.Span {
 	return json.Span{
 		TraceID:       json.TraceID(span.TraceID.String()),
 		SpanID:        json.SpanID(span.SpanID.String()),
 		Flags:         uint32(span.Flags),
 		OperationName: span.OperationName,
-		References:    fd.convertReferences(span),
-		StartTime:     model.TimeAsEpochMicroseconds(span.StartTime),
-		Duration:      model.DurationAsMicroseconds(span.Duration),
-		Tags:          fd.convertKeyValues(span.Tags),
-		Logs:          fd.convertLogs(span.Logs),
-		ProcessID:     processID,
-		Warnings:      span.Warnings,
+
+		StartTime: model.TimeAsEpochMicroseconds(span.StartTime),
+		Duration:  model.DurationAsMicroseconds(span.Duration),
+		Tags:      fd.convertKeyValuesFunc(span.Tags),
+		Logs:      fd.convertLogs(span.Logs),
 	}
 }
 
-func (fd fromDomain) convertReferences(span *model.Span) []json.Reference {
+func (fd fromDomain) convertSpan(span *model.Span, processID json.ProcessID) json.Span {
+	s := fd.convertSpanInternal(span)
+	s.ProcessID = processID
+	s.Warnings = span.Warnings
+	s.References = fd.convertReferences(span, false)
+	return s
+}
+
+func (fd fromDomain) convertSpanEmbedProcess(span *model.Span) *json.Span {
+	s := fd.convertSpanInternal(span)
+	s.Process = fd.convertProcess(span.Process)
+	s.ParentSpanID = json.SpanID(span.ParentSpanID.String())
+	s.References = fd.convertReferences(span, true)
+	return &s
+}
+
+func (fd fromDomain) convertReferences(span *model.Span, es bool) []json.Reference {
```

Comment:
* s/es/preserveParentID
 * add comment that when preserveParentID==false the parent ID is converted to a CHILD_OF reference